### ISSUE TITLE
tsmetrics cannot be found or executed

### ISSUE THREAD

**Issue by jfideler** [SonarSource/SonarTS#201]
Currently running SonarQube 2.5 with SonarQube Scanner 2.8 in a Jenkins job.

After adding the plugin "SonarTS" all our builds are failing with this error:

`node /var/lib/jenkins/workspace/xyz/Multi Branch Release Pipeline/angular2-seed/develop/angular2-seed/.sonar/sonarts-core/node_modules/tslint-sonarts/bin/tsmetrics`

here is our properties file:

# ***** Project Configuration required params ***** #
# must be unique in a given SonarQube instance
sonar.projectKey=angular2-seed
sonar.projectName=Angular2 Seed
sonar.projectVersion=1.0

# sonarqube token for sonar service account github.devtools.merrillcorp.com
sonar.login = e03e12ee1d512bad5b1a5893d24e2827dcbfeaa4
sonar.sources=src

# ***** Project Configuration optional params ***** #
# Encoding of the source code. Default is default system encoding
sonar.sourceEncoding=UTF-8
sonar.exclusions=node_modules/**, **/*.js,**/*.js.map, **/shared/mocks.**
sonar.ts.tslintconfigpath=tslint.json

# ***** Javascript settings ***** #
sonar.javascript.jQueryObjectAliases=jQuery
sonar.javascript.forceZeroCoverage=false
sonar.javascript.lcov.itReportPath=coverage
sonar.dynamicAnalysis=reuseReports
sonar.typescript.lcov.reportPaths=coverage/report-lcovonly.lscov
# ***** Web settings ***** #
sonar.web.file.suffixes=.xhtml,.rhtml,.shtml,.html

project has a dependency of       "tslint-sonarts": "^1.0.0"

**Comment by inverno** [comment]
Hello @jfideler,
there are two things that don't match up for me.
1. the property `sonar.ts.tslintconfigpath` belongs to [Pablissimo's ts plugin](https://github.com/Pablissimo/SonarTsPlugin), not SonarSource/SonarTS.
2. if you added tslint-sonarts as a dependency to your project it means that you intend to use sonarts as an extension to tslint. Also using it as a Sonar plugin is confusing.

If you decide that you want to use SonarSource/SonarTS as a SonarQube plugin then please try the following steps :

1. Stop using Pablissimo/SonarTSPlugin (remove the properties and uninstall)
2. Stop running tslint from your build script
3. Remove the tslint-sonarts dependency from your project's package.json
4. Try running sonar-scanner from `src` and let me know how it goes

If you decide that you want to use SonarSource/SonarTS as a tslint extension (you don't get metrics and complete rules descriptions in SonarQube, but you get to keep uring your tslint setup and other extensions) then go through the following steps

1. Stop using SonarSource/SonarTS as a SonarQube plugin (uninstall from SonarQube)
2. Configure your tslint.json file to use the tslint-sonarts extension, as described here : https://github.com/SonarSource/SonartS#use-in-tslint
3. Run your tslint analysis as usual and import everything with Pablissimo/SonarTSPlugin as you are probably doing now

**Comment by jfideler** [comment]
Thank you for the reply.  

The error I describe above happens irrespective of whether we have pablissimo's plugin installed.  (nonetheless, we have removed that plugin and those entries as they were from an earlier experiment).

We found our solution this morning

FWIW: It turns out that our error had nothing to do with the plug-in.  We needed to fix our Jenkins script so that it had a proper 'path' to our node environment.  (It was the 'node' part of that command that made everything blow up)
